### PR TITLE
Create new home for data files used in examples

### LINKED ISSUE #22: What should we do with all our input files?
**Abstract**

We have many input files that are just committed to the repository and others that are distributed with Cantera. We should discuss which of these should be deprecated and removed, which should be removed without deprecation, and where the files that are staying should live.

**Motivation**

In Cantera/cantera#768, we determined that several input files should be deprecated and/or removed. However, due to the size of that PR, specific discussion was deferred. This issue is where I propose to have the relevant discussion.

At present, the following input files are committed to the repository:

<details><summary>The list of files in the <code>data</code> folder</summary>
<pre>
data
├── inputs
│   ├── KOH.cti
│   ├── air.cti
│   ├── air.inp
│   ├── airNASA9.cti
│   ├── airNASA9.inp
│   ├── argon.cti
│   ├── argon.inp
│   ├── critProperties.xml
│   ├── diamond.cti
│   ├── elements.xml
│   ├── graphite.cti
│   ├── gri30.cti
│   ├── gri30.inp
│   ├── gri30_highT.cti
│   ├── gri30_ion.cti
│   ├── h2o2.cti
│   ├── h2o2.inp
│   ├── liquidvapor.cti
│   ├── lithium_ion_battery.cti
│   ├── methane_pox_on_pt.cti
│   ├── nDodecane_Reitz.cti
│   ├── nasa.cti
│   ├── nasa_condensed.cti
│   ├── nasa_gas.cti
│   ├── ohn.cti
│   ├── ptcombust.cti
│   ├── silane.cti
│   ├── silane.inp
│   ├── silicon.cti
│   ├── silicon_carbide.cti
│   ├── sofc.cti
│   └── water.cti
├── thermo
│   ├── airDataNASA9.dat
│   ├── gri30_thermo.dat
│   └── nasathermo.dat
├── transport
│   ├── gri30_tran.dat
│   └── misc_tran.dat
├── KOH.yaml
├── air.yaml
├── airNASA9.yaml
├── argon.yaml
├── diamond.yaml
├── graphite.yaml
├── gri30.yaml
├── gri30_highT.yaml
├── gri30_ion.yaml
├── h2o2.yaml
├── liquidvapor.yaml
├── lithium_ion_battery.yaml
├── methane_pox_on_pt.yaml
├── nDodecane_Reitz.yaml
├── nasa.yaml
├── nasa_condensed.yaml
├── nasa_gas.yaml
├── ohn.yaml
├── ptcombust.yaml
├── silane.yaml
├── silicon.yaml
├── silicon_carbide.yaml
├── sofc.yaml
└── water.yaml
</pre>
</details>

There are several "classes" of files here:
* CTI/YAML input files for examples (e.g., `lithium_ion_battery.cti`)
* CTI/YAML input files for data (e.g., `nasa_condensed.cti`, `liquidvapor.yaml`)
* CHEMKIN-style input files
* XML files for fundamental data (`critProperties.xml`, `elements.xml`)

Some input files might be cross some of these lines (e.g., `liquidvapor.yaml`), so the distinctions are a little fuzzy.

**Possible Solutions**

To get the discussion started, I'll propose the following changes:

1. Remove all CHEMKIN input files from the `data` folder. They are (or will shortly be) no longer needed at build time, so they will become clutter. Select files should be moved to `test/data` for tests of the conversion scripts.
2. Move input files that are used in one example to the folder with that example. IMO, the word `data` implies that the files therein are more important somehow, and moving the input files to an example-specific folder removes some of the implicit endorsement that we are giving to these files by distributing them.
  a. This is mildly complicated because several files are used in several examples across a few interfaces
3. Data files `elements.xml`, `critProperties.xml`, and `liquidvapor.yaml` remain in the `data` folder
4. All other files that are not used in an example are deprecated and removed

I'm not particularly attached to this arrangement, just throwing it out there to get something started. I'm also not particularly attached to this happening on any particular timeline, so discussion is more than welcome.

### PR DESCRIPTION
**Pull request by speth**
**Changes proposed in this pull request**

This PR is the implementation of an idea I suggested in https://github.com/Cantera/enhancements/issues/22#issuecomment-2025676862.

- Add the newly-created https://github.com/cantera/cantera-example-data repository as a Git submodule
- Depending on the value of the new SCons option `example_data`, install these data files when installing Cantera
- Refactor the automatic checkout of submodules in `SConstruct`
- Eliminate the need for the redundant copy of the data files that ends up in `build/data`

I don't regard this as the final word on which files belong in the `data` directory of the of this repo and which ones should be moved to `cantera-example-data`. So far, I've only moved those data files that were _only_ used in examples into the submodule. If we want to move any other files, it will take a bit of work to extricate them from the test suite.

**If applicable, fill in the issue number this pull request is fixing**

Closes Cantera/enhancements#22

**If applicable, provide an example illustrating new features this pull request is introducing**

Files from the `cantera-example-data` repository can be loaded by specifying the path relative to the base Cantera data directory:
```py
gas = ct.Solution('example_data/oxygen-plasma-itikawa.yaml')
```

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

File: test/python/test_utils.py [hunk 1]
```diff
@@ -322,3 +323,8 @@ def test_set_quantity(self):
         outer.set_activation_energy('cheese', 12, 'V/m')
         with pytest.raises(ct.CanteraError):
             _py_to_anymap_to_py(outer)
+
+def test_list_data_files():
+    data_files = ct.list_data_files()
+    assert "gri30.yaml" in data_files
+    assert f"example_data{os.sep}oxygen-plasma-itikawa.yaml" in data_files
```
Comment (ischoegl) [inline test/python/test_utils.py]: I’d prefer `pathlib` here

**Review by ischoegl — COMMENTED** [review]
Thanks! Minor comment but would also be interested in @bryanwweber’s input.

File: SConstruct [hunk 2]
```diff
@@ -2150,16 +2089,24 @@ elif env['OS'] == 'Darwin':
 else:
     env.PrependENVPath('LD_LIBRARY_PATH', Dir('#build/lib').abspath)
 
-for yaml in multi_glob(env, "data", "yaml"):
-    dest = Path() / "build" / "data" / yaml.name
-    build(env.Command(str(dest), yaml.path, Copy("$TARGET", "$SOURCE")))
+
+if (env["example_data"]
+    and not list(Path("data/example_data").glob("*.yaml"))
```
Comment (bryanwweber) [inline SConstruct:2094]: This short circuits the checkout if any YAML files are in the folder... I wonder if we'll run into cases where out-of-date files are used. Can we make this only conditional on the build variable?

**Review by bryanwweber — COMMENTED** [review]
Looks good to me, aside from the one minor comment!

File: SConstruct (hunk 2 above)
Comment (speth) [reply SConstruct:2094]: The current check is similar to what we do for all of the other submodules -- we're really only testing that something has been checked out, and I don't think we want to run the `git` command for every incremental build. Once the submodule has been synced at least once, managing whether it's up-to-date is left up to the user, and can be seen by whether `git status` reports uncommitted changes in the working directory.

**Review by ischoegl — APPROVED** [review]

**Review by bryanwweber — APPROVED** [review]

**Pull request merged**